packages/compass-indexes/src/modules/search-indexes.ts

```diff
@@ -18,39 +33,203 @@ export enum SearchIndexesStatuses {
-  type: ActionTypes.SetStatus;
-  status: SearchIndexesStatus;
+type SetIsRefreshingAction = {
+  type: ActionTypes.SetIsRefreshing;
+};
+
+type SetSearchIndexesAction = {
+  type: ActionTypes.SetSearchIndexes;
+  searchIndexes: SearchIndex[];
+};
+
+type SearchIndexesSortedAction = {
+  type: ActionTypes.SearchIndexesSorted;
+  searchIndexes: SearchIndex[];
+  sortOrder: SortDirection;
+  sortColumn: SearchSortColumn;
 };
 
+type SetErrorAction = {
+  type: ActionTypes.SetError;
+  error: string | null;
+};
+
+type SearchIndexesActions =
+  | SetIsRefreshingAction
+  | SetSearchIndexesAction
+  | SearchIndexesSortedAction
+  | SetErrorAction;
+
 export type State = {
   status: SearchIndexesStatus;
+  searchIndexes: SearchIndex[];
+  sortOrder: SortDirection;
+  sortColumn: SearchSortColumn;
+  error: string | null;
 };
 
 export const INITIAL_STATE: State = {
   status: 'NOT_AVAILABLE',
+  searchIndexes: [],
+  sortOrder: 'asc',
+  sortColumn: 'Name and Fields',
+  error: null,
 };
 
-export default function reducer(state = INITIAL_STATE, action: AnyAction) {
-  if (isAction<SetStatusAction>(action, ActionTypes.SetStatus)) {
+export default function reducer(
+  state = INITIAL_STATE,
+  action: AnyAction
+): State {
+  if (isAction<SetIsRefreshingAction>(action, ActionTypes.SetIsRefreshing)) {
+    return {
+      ...state,
+      status: 'REFRESHING',
+    };
+  }
+
+  if (isAction<SetSearchIndexesAction>(action, ActionTypes.SetSearchIndexes)) {
     return {
       ...state,
-      status: action.status,
+      searchIndexes: action.searchIndexes,
+      status: 'READY',
+    };
+  }
+
+  if (
+    isAction<SearchIndexesSortedAction>(action, ActionTypes.SearchIndexesSorted)
+  ) {
+    return {
+      ...state,
+      searchIndexes: action.searchIndexes,
+      sortOrder: action.sortOrder,
+      sortColumn: action.sortColumn,
+    };
+  }
+
+  if (isAction<SetErrorAction>(action, ActionTypes.SetError)) {
+    return {
+      ...state,
+      searchIndexes: [],
+      error: action.error,
+      status: 'ERROR',
     };
   }
 
   return state;
 }
 
-export const setStatus = (status: SearchIndexesStatus): SetStatusAction => ({
-  type: ActionTypes.SetStatus,
-  status,
+const setSearchIndexes = (
+  searchIndexes: SearchIndex[]
+): SetSearchIndexesAction => ({
+  type: ActionTypes.SetSearchIndexes,
+  searchIndexes,
 });
+
+const setError = (error: string | null): SetErrorAction => ({
+  type: ActionTypes.SetError,
+  error,
+});
+
+const setIsRefreshing = (): SetIsRefreshingAction => ({
+  type: ActionTypes.SetIsRefreshing,
+});
+
+export const fetchSearchIndexes = (): IndexesThunkAction<
+  Promise<void>,
+  SearchIndexesActions
+> => {
+  return async (dispatch, getState) => {
+    const {
+      isReadonlyView,
+      dataService,
+      namespace,
+      searchIndexes: { sortColumn, sortOrder, status },
+    } = getState();
+
+    if (isReadonlyView) {
+      // TODO: should we just leave the status at pending?
+      return;
+    }
```

Comment:
I think we can just return here without any status change (or maybe change the status to NOT_AVAILABLE). With `isReadonlyView` we do not show the tables, so there's no way to get here.